qcodes/instrument/base.py

```diff
@@ -409,6 +408,19 @@ def short_name(self) -> str:
         """Short name of the instrument"""
         return self._short_name
 
+    @staticmethod
+    def _is_valid_identifier(name: str) -> str:
+        """Check whether given name is a valid instrument identifier."""
+        new_name = name.replace("-", "_")
+        if not new_name.isidentifier():
+            raise ValueError(f"{name} invalid instrument identifier")
+        if name != new_name:
+            warnings.warn(
+                f"Changed {name} to {new_name} for valid instrument identifier"
+            )
```

Comment:
shall the order be changed so that if the "new name" is still not an identifier then the user sees both the original name and the "new name" and knows what happened? also, ValueError should likely include "new name", not "name" (or perhaps both)?
```suggestion
        if name != new_name:
            warnings.warn(
                f"Changed {name} to {new_name} for valid instrument identifier"
            )
        if not new_name.isidentifier():
            raise ValueError(f"{new_name} invalid instrument identifier")
```

Reply:
I agree, this makes more sense. If order is changed then "new_name" in ValueError. Having both can be confusing since "new_name" can still be the same as "name". 
I'll change the warning string to: "Changed {name} to {new_name} for instrument identifier" since it is not automatically valid, an error can still be thrown.
Thanks! I'll update it now.